sdk/tests/deqp/framework/delibs/debase/deMath.js

```diff
@@ -453,32 +453,46 @@ deMath.numberToArray = function(array, number) {
 };
 
 /**
- * Obtains the bit fragment from an array in a number
- * @param {Uint8Array} array
+ * Obtains the bit fragment from a number
+ * @param {number} x
  * @param {number} firstNdx
  * @param {number} lastNdx
  * @return {number}
  */
-deMath.getBitRange = function(array, firstNdx, lastNdx) {
-    /** @type {number} */ var bitSize = lastNdx - firstNdx;
-    /** @type {number} */ var byteSize = Math.floor(bitSize / 8) + ((bitSize % 8) > 0 ? 1 : 0);
-
-    /** @type {ArrayBuffer} */ var buffer = new ArrayBuffer(byteSize);
-    /** @type {Uint8Array} */ var outArray = new Uint8Array(buffer);
-
-    for (var bitNdx = firstNdx; bitNdx < lastNdx; bitNdx++) {
-        /** @type {number} */ var sourceByte = Math.floor(bitNdx / 8);
-        /** @type {number} */ var sourceBit = Math.floor(bitNdx % 8);
-
-        /** @type {number} */ var destByte = Math.floor((bitNdx - firstNdx) / 8);
-        /** @type {number} */ var destBit = Math.floor((bitNdx - firstNdx) % 8);
-
-        /** @type {number} */ var sourceBitValue = (array[sourceByte] & Math.pow(2, sourceBit)) != 0 ? 1 : 0;
+deMath.getBitRange = function(x, firstNdx, lastNdx) {
+    var shifted = deMath.shiftRight(x, firstNdx);
+    var bitSize = lastNdx - firstNdx;
+    var mask;
+    if (bitSize < 32)
+        mask = (1 << bitSize) -1
+    else
+        mask = Math.pow(2, bitSize) - 1;
+    var masked = deMath.binaryAnd(shifted, mask);
+    return masked;
+};
 
-        outArray[destByte] = outArray[destByte] | (Math.pow(2, destBit) * sourceBitValue);
-    }
+/**
+ * Split a large signed number into low and high 32bit dwords.
+ * @param {number} x
+ * @return {Array<number>}
+ */
+deMath.split32 = function(x) {
+    var ret = [];
+    ret[1] = Math.floor(x / 0x100000000);
+    ret[0] = x - ret[1] * 0x100000000;
+    return ret;
```

Comment:
Has this been tested with a range of inputs? I'm a little dubious it will be correct in all cases given JavaScript's promotion of all numbers to double.